src/test/sushi-integration.t.sol

```diff
@@ -627,93 +671,159 @@ contract SushiIntegrationTest is TestBase {
+    }
+
+    function test_cage_queued_dangerous_change_owner2() public {
+        execute_dangerous_timelock_action("", abi.encodeWithSelector(MasterChefLike.transferOwnership.selector, address(user2), true, false));
+        assertEq(masterchef.owner(), address(user2));
+    }
+
+    function test_cage_queued_dangerous_change_rewarder1() public {
+        execute_dangerous_timelock_action("set(uint256,uint256,address,bool)", abi.encode(join.pid(), 100, address(user2), true));
+        assertEq(masterchef.rewarder(join.pid()), address(user2));
+    }
+
+    function test_cage_queued_dangerous_change_rewarder2() public {
+        execute_dangerous_timelock_action("", abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(user2), true));
+        assertEq(masterchef.rewarder(join.pid()), address(user2));
+    }
+
+    function test_cage_queued_safe_change_diff_target() public {
+        queue_safe_timelock_action(address(user1), "setMigrator(address)", abi.encode(address(user2)), false);
+    }
+
+    function test_cage_queued_safe_change_safe_function1() public {
+        queue_safe_timelock_action(address(masterchef), "add(uint256,address,address)", abi.encode(100, address(pair), address(user2)), true);
+    }
+
+    function test_cage_queued_safe_change_safe_function2() public {
+        queue_safe_timelock_action(address(masterchef), "", abi.encodeWithSelector(MasterChefLike.add.selector, 100, address(pair), address(user2)), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args1() public {
+        queue_safe_timelock_action(address(masterchef), "set(uint256,uint256,address,bool)", abi.encode(join.pid(), 100, address(user2), false), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args2() public {
+        queue_safe_timelock_action(address(masterchef), "", abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(user2), false), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args3() public {
+        queue_safe_timelock_action(address(masterchef), "set(uint256,uint256,address,bool)", abi.encode(join.pid(), 100, address(rewarder), true), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args4() public {
+        queue_safe_timelock_action(address(masterchef), "", abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(rewarder), true), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args5() public {
+        queue_safe_timelock_action(address(masterchef), "set(uint256,uint256,address,bool)", abi.encode(join.pid() + 1, 100, address(user2), true), false);
+    }
+
+    function test_cage_queued_safe_change_safe_args6() public {
+        queue_safe_timelock_action(address(masterchef), "", abi.encodeWithSelector(MasterChefLike.set.selector, join.pid() + 1, 100, address(user2), true), false);
+    }
+
+    // Verify that we can't get around the hash comparison by providing unexpected bool values
```

Comment:
Might it be better to use fuzzing here instead, or in addition (if these specific cases are uniquely important)?

Reply:
Yeah these are uniquely important imo. Wanted to test two values < uint8 size but not 0 and 1 and the 1s bit set and not set along with the same test for > uint8 size. Anything other than 0 or 1 errors, so I'm satisfied.